### DISCUSSION TITLE
need to use save image event on click of  my custom button click function

### DISCUSSION THREAD

**Discussion by sohailamjad256** [excalidraw/excalidraw#4870] [Q&A]
Hi ...  I need to use save image event on click of  my custom button click function..how can i use your save image event ..please help..
Thanks

**Comment by sohailamjad256** [comment]
its little priority. Can i get the solution

**Comment by ad1992** [comment]
If you have a custom button then you can use the API [getFiles](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#getFiles) to get the images. 
 You can use [onChange](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#onChange) callback to autosave the images. 
 Also, can you elaborate on what issues are you facing with the package? 
Because it looks like the issues you have been facing will be easier and faster to fix If you take a look at the documentation and examples. So I would like to understand if the documentation is not clear? This will help us in making the package better!.

**Reply by sohailamjad256** [reply]
> If you have a custom button then you can use the API [getFiles](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#getFiles) to get the images. You can use [onChange](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#onChange) callback to autosave the images. Also, can you elaborate on what issues are you facing with the package? Because it looks like the issues you have been facing will be easier and faster to fix If you take a look at the documentation and examples. So I would like to understand if the documentation is not clear? This will help us in making the package better!.

yes ..i have my custom button..so.on click of  custom button i need to call that [getFiles] api and i  can save image?

**Reply by sohailamjad256** [reply]
Hi..iam attaching one image files there is one button of save image right side of export image ..so, that save image event i need to call on my click  function..
![save image](https://user-images.githubusercontent.com/53993286/156967220-5dfc88bd-2779-4436-8778-643ba6037f13.png)

**Reply by sohailamjad256** [reply]
so,how can i call..can you come up with sandbox examples...that would be helpful
